### PR TITLE
Add reference impl for sequence ops

### PR DESCRIPTION
**Pull request by BowenBao**

### CODE REVIEW

File: onnx/backend/test/case/model/sequence.py [hunk 1]
```diff
@@ -13,6 +13,64 @@
...
+    return sequence
+
+
+def SequenceLengthImpl(sequence):
+    # type: (List[np.ndarray]) -> np.int64
+    return np.int64(len(sequence))
+
+
+def SplitToSequenceImpl(tensor, split=None, axis=0, keepdims=1):
+    # type: (np.ndarray, Optional[Union[int, List[int]]], int, int) -> List[np.ndarray]
```
Comment (wschin) [inline onnx/backend/test/case/model/sequence.py]:
```suggestion
    # type: (np.ndarray, Optional[Union[int, List[int]]], int, int) -> List[np.ndarray]
```
Can we use numpy type such as `np.int64` as the `int` here?

File: onnx/backend/test/case/model/sequence.py [hunk 2]
```diff
@@ -13,6 +13,64 @@
...
+        split = 1
+        split_indices = [i * split + 1 for i in range(dim_size) if i * split + 1 < dim_size]
+        if not keepdims:
+            results = np.array_split(tensor, split_indices, axis)
+            return [np.squeeze(res, axis) for res in results]
+    if np.isscalar(split):
+        split_indices = [i * split + 1 for i in range(dim_size) if i * split + 1 < dim_size]  # type: ignore
+    else:
+        split_indices = np.cumsum(split) + 1
+    return np.array_split(tensor, split_indices, axis)  # type: ignore
```
Comment (wschin) [inline onnx/backend/test/case/model/sequence.py]: Why do you ignore the type?

File: onnx/backend/test/case/model/sequence.py [hunk 3]
```diff
@@ -13,6 +13,64 @@
 from typing import List, Optional, Text, Union
 
 
+def SequenceEmptyImpl():  # type: () -> List[Optional[np.ndarray]]
```
Comment (wschin) [inline onnx/backend/test/case/model/sequence.py]: Can we reuse those implementations to create examples in `Operators.md`?

File: onnx/backend/test/case/model/sequence.py (hunk 1 above)
Comment (BowenBao) [reply onnx/backend/test/case/model/sequence.py]: Everywhere else also uses int. I'd prefer using int here as well. In addition as below, it seems some numpy methods don't work well with mypy.

File: onnx/backend/test/case/model/sequence.py (hunk 2 above)
Comment (BowenBao) [reply onnx/backend/test/case/model/sequence.py]: some numpy methods don't work well with mypy

File: onnx/backend/test/case/model/sequence.py (hunk 3 above)
Comment (BowenBao) [reply onnx/backend/test/case/model/sequence.py]: Yes. But to create examples in Operators.md, we need to update the proto to enable Sequence Tensor Proto (in order to save test data in sequence format). That should be a separate PR of its own.

**Pull request merged**